### ISSUE TITLE
Add seek support for youtube videos

### ISSUE THREAD

**Issue by pmb6tz** [dz0ny/mopidy-youtube#34]
Hey, thanks for the great work on this plugin!

It would be awesome if we could seek forward/backward in the videos. I know this is possible in mplayer w/ffmpeg using the same stream url that the plugin uses, so I wondered if it would be possible here. 

If you want to point me in the right direction I could see about doing this myself and submitting a pull request. I mainly don't quite understand how to override the playback functionality myself.

**Comment by muhahacz** [comment]
+1

**Comment by BrainStone** [comment]
Is this an issue with Mopidy or with Mopidy-YouTube?

**Comment by natumbri** [comment]
I'm thinking about adding file caching, which passes a file on a disk to gstreamer.  Using a cached file, rather than the file direct from YouTube might solve any seeking issues.

New branch here: [feature/cache](/natumbri/mopidy-youtube/tree/feature/cache)

Config option required:
```
allow_cache = true
```

Feedback appreciated.

**Comment by natumbri** [comment]
@pmb6tz @lukasmrtvy @BrainStone is anyone interested in testing this? I've merged into [develop](/natumbri/mopidy-youtube/tree/develop).  Would be interested to hear if it helps, or if you're no longer interested (in which case I'll close the issue). 

Cheers.

**Comment by BrainStone** [comment]
I currently don't use this software anymore. But if I would I'd still be interested, fyi.

**Comment by kingosticks** [comment]
Did you ever try https://docs.mopidy.com/en/latest/api/backend/?highlight=should_download#mopidy.backend.PlaybackProvider.should_download for this @natumbri ?

**Comment by natumbri** [comment]
That's a good question @kingosticks! Somehow it ended up in the [PlaybackProvider](https://github.com/natumbri/mopidy-youtube/blob/a3e44b2542d386a395795e191ac122adf3355398/mopidy_youtube/backend.py#L412) but I'm not convinced that I ever tested it, or even thought carefully about whether it is a good idea for this extension.

**Comment by kingosticks** [comment]
I would have thought it'd make an impact (good or bad) for an issue like this.

**Comment by natumbri** [comment]
Yeah, absolutely it seems relevant. I will have a look at it - thanks for raising.
